### PR TITLE
PV Encryption via UI

### PR DESCRIPTION
**Pull request by am-agrawa**

### CODE REVIEW

File: ocs_ci/ocs/ui/helpers_ui.py [hunk 1]
```diff
@@ -95,6 +99,171 @@ def format_locator(locator, string_to_insert):
...
+        base_ui_obj.do_send_keys(pvc_loc["tls-server-name"], tls_server_name)
+        logger.info("Enter Vault Enterprise Namespace")
+        base_ui_obj.do_clear(pvc_loc["vault-enterprise-namespace"])
+        base_ui_obj.do_send_keys(pvc_loc["vault-enterprise-namespace"], vault_namespace)
+        logger.info("Selecting CA Certificate")
+        # print(os.path.abspath(constants.VAULT_CA_CERT_PEM))
+        # logger.info(os.path.abspath(constants.VAULT_CA_CERT_PEM))
+        # element = base_ui_obj.driver.find_element_by_xpath("//*[@id='modal-container']/div/div/div/form/div/div[2]/div/div/div[4]/div[2]/div/input")
+        # base_ui_obj.driver.execute_script("arguments[0].value='typed';",element)
+        base_ui_obj.send_keys(pvc_loc["browse-ca-certificate"], text=os.path.abspath(constants.VAULT_CA_CERT_PEM))
```
Comment (afreen23) [inline ocs_ci/ocs/ui/helpers_ui.py:211]: I cannot find any reference for `base_ui_obj.send_keys` .
https://github.com/red-hat-storage/ocs-ci/blob/37613fb3cf8d076a8d80787258a4307d0633570f/ocs_ci/ocs/ui/base_ui.py#L330

**Review by afreen23 — CHANGES_REQUESTED** [review]

File: ocs_ci/ocs/ui/helpers_ui.py (hunk 1 above)
Comment (afreen23) [reply ocs_ci/ocs/ui/helpers_ui.py:211]: Why not doing something like  ? :
```python
file_input = driver.find_element(By.XPATH, "(//input[@type='file'])[1]")
file_input.send_keys('<your location>')
```

The `send_keys` function in selenium accepts text and also there is no such function on the `base_ui_obj` object of class `PageNavigator`.

Not too familiar with this code base but just cant find anything.
Also `do_send_keys` function wont work here due to that fact it waits for an element to be clickable, just guessing if you wanted to use that.

Comment (am-agrawa) [reply ocs_ci/ocs/ui/helpers_ui.py]: @afreen23 Thanks, it's working now. :)

File: ocs_ci/ocs/ui/helpers_ui.py [hunk 2]
```diff
@@ -95,6 +98,157 @@ def format_locator(locator, string_to_insert):
...
+    logger.info("Storage Class Description")
+    base_ui_obj.do_send_keys(pvc_loc["sc-description"], "this is a test storage class")
+    logger.info("Storage Class Reclaim Policy")
+    base_ui_obj.do_click(pvc_loc["reclaim-policy"])
+    if reclaim_policy == "Delete":
+        base_ui_obj.do_click(pvc_loc["reclaim-policy-delete"])
+    elif reclaim_policy == "Retain":
+        base_ui_obj.do_click(pvc_loc["reclaim-policy-retain"])
+
+    if ocp_version >= "4.9":
```
Comment (rachael-george) [inline ocs_ci/ocs/ui/helpers_ui.py:153]: Would it be possible to be compare the version as mentioned here: https://github.com/red-hat-storage/ocs-ci/pull/4872 ?
This would do away with any irregularities there might be when the version is 4.10 or higher.

File: ocs_ci/ocs/ui/helpers_ui.py [hunk 3]
```diff
@@ -95,6 +98,157 @@ def format_locator(locator, string_to_insert):
...
+
+    logger.info("Storage Class Provisioner")
+    base_ui_obj.do_click(pvc_loc["provisioner"])
+    if provisioner == "rbd":
+        base_ui_obj.do_click(pvc_loc["rbd-provisioner"])
+    elif provisioner == "cephfs":
+        base_ui_obj.do_click(pvc_loc["cephfs-provisioner"])
+    logger.info("Storage Class Storage Pool")
+    base_ui_obj.do_click(pvc_loc["storage-pool"])
+    base_ui_obj.do_click(pvc_loc["ceph-block-pool"])
```
Comment (rachael-george) [inline ocs_ci/ocs/ui/helpers_ui.py:171]: It would be good to also have the option to create a new CBP, but that can be taken up as part of a different PR.

File: ocs_ci/ocs/ui/helpers_ui.py [hunk 4]
```diff
@@ -95,6 +98,157 @@ def format_locator(locator, string_to_insert):
...
+        logger.info("Storage Class with Encryption")
+        base_ui_obj.do_click(pvc_loc["encryption"])
+        logger.info("Checking if 'Change connection details' option is available")
+        conn_details = base_ui_obj.check_element_text(
+            expected_text="Change connection details"
+        )
+        if conn_details:
+            logger.info("Click on Change Connection Details")
+            base_ui_obj.do_click(pvc_loc["connections-details"])
+        if ocp_version >= "4.9":
```
Comment (rachael-george) [inline ocs_ci/ocs/ui/helpers_ui.py:182]: There is also the option of selecting an existing KMS connection detail from the dropdown menu.

File: ocs_ci/ocs/ui/helpers_ui.py [hunk 5]
```diff
@@ -95,6 +98,157 @@ def format_locator(locator, string_to_insert):
...
+            logger.info("Click on Create new KMS connection")
+            base_ui_obj.do_click(pvc_loc["new_kms"])
+        logger.info("Storage Class Service Name")
```
Comment (rachael-george) [inline ocs_ci/ocs/ui/helpers_ui.py:185]: By service name, do you mean the "Connection name" text box in the UI? If yes, that is the KMS service name

File: ocs_ci/ocs/ui/helpers_ui.py [hunk 6]
```diff
@@ -95,6 +98,157 @@ def format_locator(locator, string_to_insert):
...
+        base_ui_obj.do_clear(pvc_loc["service-name"])
+        base_ui_obj.do_send_keys(pvc_loc["service-name"], service_name)
+        logger.info("Storage Class Address")
```
Comment (rachael-george) [inline ocs_ci/ocs/ui/helpers_ui.py:188]: It is the address of the vault node

File: ocs_ci/ocs/ui/helpers_ui.py [hunk 7]
```diff
@@ -95,6 +98,157 @@ def format_locator(locator, string_to_insert):
...
+        base_ui_obj.do_clear(pvc_loc["kms-address"])
+        base_ui_obj.do_send_keys(pvc_loc["kms-address"], kms_address)
+        logger.info("Storage Class Port")
```
Comment (rachael-george) [inline ocs_ci/ocs/ui/helpers_ui.py:191]: Vault port

File: ocs_ci/ocs/ui/helpers_ui.py [hunk 8]
```diff
@@ -95,6 +98,157 @@ def format_locator(locator, string_to_insert):
...
+        )
+        client_cert_pem.send_keys(os.path.abspath(constants.VAULT_CLIENT_CERT_PEM))
+        logger.info("Selecting Client Private Key")
+        client_private_key_pem = base_ui_obj.driver.find_element(
+            By.XPATH, "(//input[@type='file'])[3]"
+        )
+        client_private_key_pem.send_keys(os.path.abspath(constants.VAULT_PRIVKEY_PEM))
+        base_ui_obj.take_screenshot()
+        logger.info("Saving Key Management Service Advanced Settings")
+        base_ui_obj.do_click(pvc_loc["save-advanced-settings"], enable_screenshot=True)
```
Comment (rachael-george) [inline ocs_ci/ocs/ui/helpers_ui.py:222]: Since these are optional values, it would be good to add conditions to use/set them

File: tests/ui/test_pv_encryption_ui.py [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+            logger.info(f"Waiting for '{pvc.name}' pvc deletion")
+            pvcs[0].ocp.wait_for_delete(pvc.name, timeout=120)
+
+        logger.info("Checking if pvc's are deleted or not")
+        if len(pvcs) > 0:
+            assert f"PVC {pvcs[0].name} does not deleted"
+
+        # Verify whether the key is deleted in Vault. Skip check for kv-v2 due to BZ#1979244
+        logger.info(
+            "Verify whether the key is deleted in Vault. Skip check for kv-v2 due to BZ#1979244"
```
Comment (rachael-george) [inline tests/ui/test_pv_encryption_ui.py:279]: This was fixed in 4.9

**Review by afreen23 — APPROVED** [review]

File: ocs_ci/ocs/ui/helpers_ui.py (hunk 2 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/helpers_ui.py:153]: Ack
